huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go

```diff
@@ -485,73 +563,119 @@ func updatePolicy(d *schema.ResourceData, client *golangsdk.ServiceClient) error
 			PolicyID: newP.(string),
 		}).Extract()
 		if err != nil {
-			return fmtp.Errorf("Error binding policy to vault: %s", err)
+			return fmt.Errorf("error binding policy to vault: %s", err)
 		}
 	} else {
 		_, err := vaults.UnbindPolicy(client, d.Id(), vaults.BindPolicyOpts{
 			PolicyID: oldP.(string),
 		}).Extract()
 		if err != nil {
-			return fmtp.Errorf("Error unbinding policy from vault: %s", err)
+			return fmt.Errorf("error unbinding policy from vault: %s", err)
 		}
 	}
 	return nil
 }
 
-func resourceCBRVaultV3Update(d *schema.ResourceData, meta interface{}) error {
+func resourceVaultUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
 	client, err := config.CbrV3Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating Huaweicloud CBR v3 client: %s", err)
+		return diag.Errorf("error creating CBR v3 client: %s", err)
 	}
 
-	if d.HasChanges("name", "size", "auto_expand") {
-		opts := vaults.UpdateOpts{}
+	opts := vaults.UpdateOpts{}
+	if d.HasChange("name") {
 		opts.Name = d.Get("name").(string)
-		opts.Billing = &vaults.BillingUpdate{
-			Size: d.Get("size").(int),
+	}
+
+	if isPrePaid(d) {
+		if d.HasChange("size") {
+			return diag.Errorf("the prepaid vault do not support the size updation")
```

Comment:
’updation‘ is not a correct word, please use 'update' or 'updating'

Reply:
the description has been updated.